multi-agent/adapters/inbound/flask/endpoints/_collaboration_shared.py

```
@@ -0,0 +1,114 @@
+shared by ``builtins.py`` (admin-only built-in routes) and ``resources.py``
+(the generic resource CRUD routes, which admins also use to mutate
+built-in resources via ``guard_write_access``) so both enforce the same
+cooperative lock instead of only the former.
+
+``guard_write_access_with_lock`` combines the ownership/admin check with
+the cooperative lock check for the generic resource CRUD routes —
+``resources.py``'s ``update_resource`` and ``delete_resource`` both need
+"authorize the mutation, then reject if a built-in is locked by another
+admin" and previously duplicated that combination inline.
+"""
+from typing import Any, Dict, Optional, Tuple
```

Ruff (0.15.21)

[warning] `typing.Dict` is deprecated, use `dict` instead

(UP035)